plugin/cache.go

```diff
@@ -163,14 +167,56 @@ func CachedList(ctx context.Context, g Group) ([]Entry, error) {
 			entry.setID(id)
 		}
 
-		return entries, nil
+		return searchedEntries, nil
 	})
 
 	if err != nil {
 		return nil, err
 	}
 
-	return cachedEntries.([]Entry), nil
+	return cachedEntries.(map[string]Entry), nil
+}
+
+// CachedListReadOnly returns the current value of CachedList(g) that's in
+// the cache, and a Boolean value that's true if the value exists,
+// false otherwise. The returned error is the error returned by the
+// last CachedList(g) call (since errors are also cached), so callers
+// should check the Boolean value first before checking the error.
+//
+// CachedListReadOnly is primarily needed by the FUSE layer because it is
+// optimized for frequent, concurrent reads. It should not be used anywhere
+// else.
+//
+// NOTE: If the entry's disabled caching for plugin.ListOp, then CachedListReadOnly
+// will return g.List(context.Background()) with the boolean set to true.
+// This is to optimize the FUSE layer by avoiding unnecessary lock acquisitons.
+func CachedListReadOnly(g Group) (map[string]Entry, bool, error) {
```

Comment:
This is kind of a weird name, but the other option I considered (`GetCachedList`) also sounded weird.

Also, since this method's only needed by FUSE, would it be worth moving the FUSE package back into plugin so that this isn't included in the godoc? Just a thought.